microcosm_flask/swagger/definitions.py

```diff
@@ -157,19 +186,34 @@ def build_path_for_integer_param(ns, operation, arguments: Set):
     return path
 
 
-def build_path(operation, ns):
+def create_uri_templates(arguments):
+    return {
+        argument: f"{{{argument}}}"
+        for argument in arguments
+    }
+
+
+def build_path(operation: Operation, ns: Namespace, schema_request_arguments: Optional[Dict[Tuple, List[str]]] = None):
+    expected_arguments: List[str] = (
+        schema_request_arguments.get((ns.endpoint_for(operation), operation.value.method), [])
+        if schema_request_arguments
+        else []
+    )
+
+    expected_uri_templates = create_uri_templates(expected_arguments)
+
     try:
-        return ns.url_for(operation, _external=False)
-    except BuildError as error:
+        # flask will sometimes try to quote '{' and '}' characters
+        return unquote(ns.url_for(operation, _external=False, **expected_uri_templates))
+    except (BuildError, ValueError) as error:
```

Comment:
This section mainly exists to cover any bases that may have been missed in the above check. It may be the case that ValueError (i.e. from the int-based part argument) is the only one that's really required, in which case, I can just remove the fallback and just handle that case here.

Reply:
Scratch that, there are some endpoints (e.g. alias, upload_for) where the pre-fetching strategy fails. I could dig into that more, or just allow the fallback to keep existing here.